### DISCUSSION TITLE
how to persist a ta4j strategy to database(file/json or any persistant data source)

### DISCUSSION THREAD

**Discussion by vh1ne** [ta4j/ta4j#1161] [General]
Hi,
I am currently developing a trading software for my personal use using ta4j. I want to save strategy paramertes to persistant data source and retreive 
can anyone please suggest a way to acheive this.
Thanks

**Comment by sgflt** [comment]
You can simply load dat from database and pass them to Strategy Factory.

Like:
```java
@Component
class BuyStrategyFactory {

  // TODO
  Rule createEntryRule(final BarSeries series, final Map<String, Number> config) {
    final var closePrice = NumericIndicator.closePrice(series);
    final var smaIndicator = closePrice.sma(config.getOrDefault("entrySmaBarCount", 23).intValue());
  ...
  }

...
```

**Reply by marsteff** [reply]
I've done it with the composite pattern and reflection. So I can create a strategy completely in the database and the code don't know, which Indicatores I am using.

**Comment by TheCookieLab** [comment]
I have a full solution to strategy/rule serialization/deserialization in my
personal/private fork of ta4j. The codebases diverged years ago but I'm in
the process of porting this feature (among many) back to Ta4j.
